src/main/java/com/google/devtools/build/remote/client/RemoteClient.java

```diff
@@ -173,21 +175,40 @@ private static void printList(List<String> list, int limit) {
     }
   }
 
+  private static Digest toV2(com.google.devtools.remoteexecution.v1test.Digest d) {
+    return Digest.newBuilder().setHash(d.getHash()).setSizeBytes(d.getSizeBytes()).build();
+  }
+
+  private static Platform toV2(com.google.devtools.remoteexecution.v1test.Platform p) {
+    // return Digest.newBuilder().setHash(d.getHash()).setSizeBytes(d.getSizeBytes()).build();
+    Platform.Builder builder = Platform.newBuilder();
+    for (com.google.devtools.remoteexecution.v1test.Platform.Property property :
+        p.getPropertiesList()) {
+      builder.addProperties(
+          Platform.Property.newBuilder().setName(property.getName()).setValue(property.getValue()));
+    }
+    return builder.build();
+  }
+
   // Output for print action command.
-  private void printAction(Action action, int limit) throws IOException {
+  private void printActionV1(com.google.devtools.remoteexecution.v1test.Action action, int limit)
+      throws IOException {
+    // Note: Command V2 is backward compatible to V1. It adds fields but does not remove them, so we
+    // can use it here.
     Command command;
```

Comment:
Extract these lines into a getCommand function to match the new getAction function, you can then reuse it in printAction and I think also in setupDocker.